File: src/main/java/ru/arc/ai/Conversation.java
package ru.arc.ai;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bukkit.Location;

import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Conversation {

    UUID playerUuid;
    Location location;
    double radius;
    String gptId;
    String archetype;
    long lastMessageTime;
    long lifeTime;
    String talkerName;
    Integer npcId;
    String endMessage;
    @Builder.Default
    boolean privateConversation = true;

}

File: src/main/kotlin/ru/arc/ai/Conversation.kt
package ru.arc.ai

import org.bukkit.Location
import java.util.UUID

data class Conversation(
    val playerUuid: UUID? = null,
    val location: Location? = null,
    val radius: Double = 0.0,
    val gptId: String? = null,
    val archetype: String? = null,
    var lastMessageTime: Long = 0L,
    val lifeTime: Long = 0L,
    val talkerName: String? = null,
    val npcId: Int? = null,
    val endMessage: String? = null,
    val privateConversation: Boolean = true
)
